File: app/src/main/java/com/ikvych/cocktail/data/repository/DrinkDbRepositoryImpl.java
package com.ikvych.cocktail.data.repository;

import android.content.Context;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import com.ikvych.cocktail.data.database.DrinkDao;
import com.ikvych.cocktail.data.database.DrinkDataBase;
import com.ikvych.cocktail.data.entity.Drink;
import com.ikvych.cocktail.data.repository.base.DrinkDbRepository;

import java.util.List;

public class DrinkDbRepositoryImpl implements DrinkDbRepository {

    private DrinkDao drinkDao;

    public DrinkDbRepositoryImpl(Context context) {
        drinkDao = DrinkDataBase.getInstance(context).getDrinkDao();
    }

    @Override
    public LiveData<List<Drink>> getDrinks() {
        return drinkDao.getAllDrink();
    }


    @Override
    public void saveDrink(Drink drink) {
        new SaveDrinkAsyncTask(drinkDao).execute(drink);
    }

    private static class SaveDrinkAsyncTask extends AsyncTask<Drink, Void, Void> {

        private DrinkDao drinkDao;

        SaveDrinkAsyncTask(DrinkDao drinkDao) {
            this.drinkDao = drinkDao;
        }

        @Override
        protected Void doInBackground(Drink... drinks) {
            for (Drink drink : drinks) {
                drinkDao.insert(drink);
            }
            return null;
        }
    }
}

File: app/src/main/java/com/ikvych/cocktail/data/repository/DrinkDbRepositoryImpl.kt
package com.ikvych.cocktail.data.repository

import android.content.Context
import android.os.AsyncTask
import androidx.lifecycle.LiveData
import com.ikvych.cocktail.data.database.DrinkDao
import com.ikvych.cocktail.data.database.DrinkDataBase
import com.ikvych.cocktail.data.entity.Drink
import com.ikvych.cocktail.data.repository.base.DrinkDbRepository


class DrinkDbRepositoryImpl(context: Context):
    DrinkDbRepository {

    private val drinkDao: DrinkDao = DrinkDataBase.getInstance(context)!!.drinkDao()


    override fun getDrinks(): LiveData<List<Drink>> {
        return drinkDao.getAllDrinks()
    }

    override fun getFavoriteDrinks(): LiveData<List<Drink>> {
        return drinkDao.getAllFavoriteDrinks()
    }

    override fun saveDrink(drink: Drink) {
        SaveDrinkAsyncTask(drinkDao).execute(drink)
    }

    override fun findDrinkById(drinkId: Long): Drink {
        return FindDrinkAsyncTask(drinkDao).execute(drinkId).get()
    }

    override fun findDrinkByName(drinkName: String): Drink {
        return FindDrinkByNameAsyncTask(drinkDao).execute(drinkName).get()
    }

    private companion object class SaveDrinkAsyncTask(private val drinkDao: DrinkDao): AsyncTask<Drink, Unit, Unit>() {

        override fun doInBackground(vararg params: Drink) {
            params.forEach { drink ->  drinkDao.insert(drink)}
        }
    }
}